lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -0,0 +1,485 @@
+        df->setMinimumSignificantDigits(minSigDigits);
+        df->setMaximumSignificantDigits(maxSigDigits);
+        return S_OK;
+    }
+
+    HRESULT SetNumberFormatIntFracDigits(IPlatformAgnosticResource *resource, const uint16 minFracDigits, const uint16 maxFracDigits, const uint16 minIntDigits)
+    {
+        auto *numFormatResource = reinterpret_cast<PlatformAgnosticIntlObject<icu::NumberFormat> *>(resource);
+        icu::NumberFormat *nf = numFormatResource->GetInstance();
+        nf->setMinimumIntegerDigits(minIntDigits);
+        nf->setMinimumFractionDigits(minFracDigits);
+        nf->setMaximumFractionDigits(maxFracDigits);
+        return S_OK;
+    }
+
+    // We explicitly declare these specializations of FormatNumber so the compiler creates them
+    // because they will be used in another compilation unit,
+    // at which time we cannot generate code for specializations of this template.
+    template const char16 *FormatNumber<>(IPlatformAgnosticResource *formatter, const int32_t val, const uint16 formatterToUse, const uint16 currencyDisplay, const char16 *currencyCode);
+    template const char16 *FormatNumber<>(IPlatformAgnosticResource *formatter, const double val, const uint16 formatterToUse, const uint16 currencyDisplay, const char16 *currencyCode);
+
+    template <typename T>
+    const char16 *FormatNumber(IPlatformAgnosticResource *formatter, const T val, const uint16 formatterToUse, const uint16 currencyDisplay, const char16 *currencyCode)
+    {
+        icu::UnicodeString result;
+
+        auto *formatterHolder = reinterpret_cast<PlatformAgnosticIntlObject<icu::NumberFormat> *>(formatter);
+        if (!formatterHolder)
+        {
+            AssertOrFailFastMsg(formatterHolder, "Formatter did not hold an object of type `FinalizableIntlObject<icu::NumberFormat>`");
+            return nullptr;
+        }
+
+        icu::NumberFormat *numberFormatter = formatterHolder->GetInstance();
+
+        // __formatterToUse: 0 (default) is number, 1 is percent, 2 is currency
+        if (formatterToUse == 0 || formatterToUse == 1)
+        {
+            numberFormatter->format(val, result);
+        }
+        else if (formatterToUse == 2)
+        {
+            UErrorCode error = UErrorCode::U_ZERO_ERROR;
+
+            // const UChar *currencyCode = reinterpret_cast<const UChar *>(numberFormatter->getCurrency());
+            const UChar *uCurrencyCode = reinterpret_cast<const UChar *>(currencyCode);
+            const char *localeName = numberFormatter->getLocale(ULocDataLocaleType::ULOC_ACTUAL_LOCALE, error).getName();
+
+            if (U_FAILURE(error))
+            {
+                AssertMsg(false, "numberFormatter->getLocale failed");
+            }
+
+            UBool isChoiceFormat = false;
+            int32_t currencyNameLen = 0;
+
+            // __currencyDisplayToUse: 0 (default) is symbol, 1 is code, 2 is name
+            if (currencyDisplay == 0) // symbol ($42.00)
```

Comment:
Replace magic numbers with an enum?

Reply:
Will do

Reply:
(These numbers are also used on the JS side -- will mirror the enum over there too.)